src/hotspot/share/opto/node.cpp

```diff
@@ -568,6 +568,10 @@ Node *Node::clone() const {
   if (n->is_SafePoint()) {
     n->as_SafePoint()->clone_replaced_nodes();
   }
+  if (n->is_Halt()) {
```

Comment:
This line seems unnecessary. I see HaltNode::size() increases from 0x60 to 0x70 with your patch.  Memcpy at line 502 covers it, doesn't it?

Reply:
`_reachable` and `_halt_reason` are available on `HaltNode`, not `Node`,  so I think this condition is necessary.
L502 uses the result of `Node::size_of()`, and I overrided it in `HaltNode`. So memory allocation at L501 and memcpy at L502 are safety.

Reply:
size_of() is a virtual function.  line 500 should use HaltNode::size_of(), which give you 112(0x70) bytes, right?
I mean memcpy at line 502 should cover HaltNode's member variables.  
`Copy::conjoint_words_to_lower((HeapWord*)this, (HeapWord*)n, s);`

Reply:
Ah, you are right, thanks!
I reverted the change for node.cpp.